### ISSUE TITLE
Questions about model training

### ISSUE THREAD

**Issue by jonDel** [RasaHQ/rasa_core#823]
**Rasa Core version**:
rasa-core           0.9.8

**Python version**: 
Python 3.6.6 :: Anaconda, Inc.

**Operating system** (windows, osx, ...):
Fedora Linux

**Issue**:
Considering rasa default deep learning model, what is the size/proportion to training data of:
- validation set:
- test set?
Is there an early stopping strategy, or the training session just goes as long as the pre-defined number of epochs (no specific treatment to prevent overfitting)?

**Comment by akelad** [comment]
There's no early stopping currently. As for validation set, the standard is 10% https://github.com/RasaHQ/rasa_core/blob/master/rasa_core/train.py#L56
If you want to test on a test set this is something you have to define yourself

**Comment by michaelcapizzi** [comment]
@jonDel you can implement a `keras` `callback` for early stopping and add it to the `kwargs` of `agent.train()`:

for example:
```
ES_CALLBACK = EarlyStopping(
    monitor="val_acc",              # monitor validation accuracy
    mode="max",
    min_delta=0.002,                # if no increase of more than 0.002
    patience=40,                    # over 20 epochs
)
```

**Comment by michaelcapizzi** [comment]
In our case, we max out on `val_acc=1.0` after ~45 epochs, but the defaults are 100.  However, I wonder if, in the case of a dialog manager, trained on stories that we create, if "overfitting" in this case is actually desirable.  In other words, assuming that...

(1) you have generated a relatively exhaustive list of possible stories
(2) you allow `rasa_core` to augment the data beyond that

...perhaps "overfitting" in the traditional sense is less of a problem because the bot will likely *not* see a novel case at all.

I'm not sure I believe this yet....just considering it.  :)
